src/shapes/src/SmoothCorner.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2010-2019 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <boost/algorithm/clamp.hpp>
+
+#include <shapes/SmoothCorner.hpp>
+#include <utils/Vector.hpp>
+#include <utils/math/abs.hpp>
+#include <utils/math/coordinate_transformation.hpp>
+#include <utils/math/vec_rotate.hpp>
+
+namespace Shapes {
+
+void SmoothCorner::calculate_dist(const Utils::Vector3d &pos, double &dist,
+                                  Utils::Vector3d &vec) const {
+
+  auto pos_shifted = pos - m_center;
+  auto rel_z = pos_shifted * m_axis;
+
+  // first, do only the 2d problem, add the axis projection later
+  auto pos_projected = pos_shifted - m_axis * rel_z;
+  auto pos_norm = pos_projected.norm();
+
+  if (pos_norm == 0.) {
+    dist = m_radius;
+    vec = -dist * m_direction;
+  } else {
+    // go to cylindrical coordinates (r,delta_phi) where m_axis is the z-axis
+    // and  m_direction defines the delta_phi = 0 axis
+    auto x = pos_projected * m_direction;
+    auto y = pos_projected * Utils::vector_product(m_axis, m_direction);
+    auto pos_delta_phi = std::atan2(y, x);
+    while (pos_delta_phi > 2 * Utils::pi()) {
+      pos_delta_phi -= Utils::pi();
+    }
```

Comment:
Why do we need to do this? According to [std::atan2](https://en.cppreference.com/w/cpp/numeric/math/atan2) the return value should be between [-pi, pi] if nothing goes wrong. If it is necessary, is [std::fmod](https://en.cppreference.com/w/cpp/numeric/math/fmod) what you are looking for?

Reply:
Thanks for pointing that out, I thought the result would be in [0,2 pi]. I will remove the loop